traversal/selector/exploreInterpretAs.go

```diff
@@ -0,0 +1,65 @@
+package selector
+
+import (
+	"fmt"
+
+	"github.com/ipld/go-ipld-prime/datamodel"
+)
+
+type ExploreInterpretAs struct {
+	next Selector // selector for element we're interested in
+	adl  string   // reifier for the ADL we're interested in
+}
+
+// Interests for ExploreIndex is just the index specified by the selector node
+func (s ExploreInterpretAs) Interests() []datamodel.PathSegment {
+	return s.next.Interests()
+}
+
+// Explore returns the node's selector if
+// the path matches the index for this selector or nil if not
+func (s ExploreInterpretAs) Explore(n datamodel.Node, p datamodel.PathSegment) (Selector, error) {
+	return s.next, nil
+}
+
+// Decide always returns false because this is not a matcher
+func (s ExploreInterpretAs) Decide(n datamodel.Node) bool {
+	return false
+}
+
+// NamedReifier indicates how this selector expects to Reify the current datamodel.Node.
+func (s ExploreInterpretAs) NamedReifier() string {
+	return s.adl
+}
+
+// Reifiable provides a feature detection interface on selectors to understand when
+// and if Reification of the datamodel.node should be attempted when performing traversals.
+type Reifiable interface {
+	NamedReifier() string
+}
```

Comment:
(Reviewing out loud, not a change request)  This interface is implemented only once and used in only one place, so I wonder if it's necessary.  Since the set of types for the compiled selectors[^1] is considered closed, it seems like we could just as well use a concrete type switch?

I'm okay leaving it this way too, because it certainly works, and I don't think it gets in the way, and perhaps someone has an idea that there will be more than one selector clause in the future which asks for a reification to be performed.

[^1] - (I wish we had done https://github.com/ipld/go-ipld-prime/issues/236 by now so I could refer to what I mean by `selectorengine.*`...)